charmhelpers/contrib/openstack/utils.py

```diff
@@ -1207,12 +1208,19 @@ def _ows_check_services_running(services, ports):
     return ows_check_services_running(services, ports)
 
 
-def ows_check_services_running(services, ports):
+def ows_check_services_running(services, ports, use_ssl_check=False,
+                               ssl_check_info=None):
     """Check that the services that should be running are actually running
     and that any ports specified are being listened to.
 
     @param services: list of strings OR dictionary specifying services/ports
     @param ports: list of ports
+    @param use_ssl_check: Set to True if you want to use ssl to check
+                              service ports rather than netcat. Default is
+                              False.
+    @param ssl_check_info: If use_ssl_check is True this is a dict of
+                           {key: <path to key>, cert: <path to cert>} used
+                           by ssl when checking ports using SSL.
```

Comment:
Just a thought here: It might be worth to create class/namedtuple/dataclass for this parameter because the way it's unpacked into arguments down the line (`**ssl_check_info`). If users provides bad dict, it will lead to `TypeError`s  with unexpected args or missing args.

Reply:
fair enough i can fix that.